src/Infrastructure/LoggingPaymentNotificationVerifier.php

```diff
@@ -30,8 +30,13 @@ public function verify( array $request ) {
 		try {
 			$this->verifier->verify( $request );
 		} catch ( PayPalPaymentNotificationVerifierException $exception ) {
-			$this->logger->log( $this->logLevel, $exception->getMessage(), [ self::CONTEXT_EXCEPTION_KEY => $exception ] );
-			$this->logger->log( LogLevel::DEBUG, 'Paypal request data', $request );
+			if ( $exception->getCode() == PayPalPaymentNotificationVerifierException::ERROR_UNSUPPORTED_STATUS ) {
+				$this->logger->log( LogLevel::INFO, 'Unsupported payment_status', $request );
```

Comment:
The IPN notification endpoint is expected to at least receive all `payment_status` messages from PPL and return a "200 OK". It does not need to do something with them. Until the FR people request more status handled, we just ignore them.

The exception thing is a bit weird, I agree. The alternative would be to check the status in the route handler and create a Response before calling the veryfier. That way, we could even implement status-based routing later if we need it (I know, YAGNI). Go ahead if you like.